src/test_suite/suite_filter.ts

```diff
@@ -0,0 +1,124 @@
+import { PeekableSequence } from './peekable_sequence';
+
+type SuitePredicate = (suites: string[]) => boolean;
+
+// Constructs a SuitePredicate from a a textual boolean expression
+// over suite names. The expression can be made up of
+//   TERM: a suite name - any sequence of non-space characters that
+//         does not include the symbols '(', ')', '&', '|', and '!'
+//   LOGICAL OR: '|'
+//   LOGICAL AND: '&'
+//   LOGICAL NEGATION: '!'
+//   PARENTHESES: '(', ')'
+//
+// The SuitePredicate takes a single parameter that is an array of
+// TERMS, each of which is considered to have a true value. It returns
```

Comment:
Just for my understanding, what do you mean by this? "each of which is considered to have a true value."; what does it mean for a term to have a truth value? Can you give me an example?

Reply:
I changed the comment to
~~~
// Constructs a SuitePredicate from a a textual boolean expression
// over suite names. The expression can be made up of
//   TERM: a suite name - any sequence of non-space characters that
//         does not include the symbols '(', ')', '&', '|', and '!'
//   LOGICAL OR: '|'
//   LOGICAL AND: '&'
//   LOGICAL NEGATION: '!'
//   PARENTHESES: '(', ')'
//
// The SuitePredicate takes a single parameter that is an array of
// TERMS whose values are considered to be true. All other terms (those
// that don't appear on the list) are considered to be false. The
// SuitePredicate returns the truth value of the expression.
//
// So, for example, if the suites parameter of the SuitePredicate was
//   ['a', 'b']
// then the following expressions would evaluate to true:
//   'a'
//   'b'
//   'a & b'
// and the following expressions would evaluate to false:
//   'x'
//   '!a'
//
~~~

Reply:
Makes sense, thanks!